awscli/customizations/overridesslcommonname.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import os
+
+DEFAULT_DNS_SUFFIX = 'amazonaws.com'
+CHINA_DNS_SUFFIX = 'amazonaws.com.cn'
+GOV_DNS_SUFFIX = 'sc2s.sgov.gov'
+
+common_regions = [
+    "af-south-1",
+    "ap-east-1",
+    "ap-northeast-1",
+    "ap-northeast-2",
+    "ap-northeast-3",
+    "ap-south-1",
+    "ap-southeast-1",
+    "ap-southeast-2",
+    "ap-southeast-3",
+    "ca-central-1",
+    "eu-north-1",
+    "eu-south-1",
+    "eu-west-1",
+    "eu-west-2",
+    "eu-west-3",
+    "fips-us-east-1",
+    "fips-us-east-2",
+    "fips-us-west-1",
+    "fips-us-west-2",
+    "me-south-1",
+    "sa-east-1",
+    "us-east-2",
+    "us-west-1",
+    "us-west-2",
+]
+
+common_regions_with_dns_suffix = {
+    region: DEFAULT_DNS_SUFFIX for region in common_regions
+}
+
+sslCommonNameServiceRegions = {
+    "sqs": {
+        "us-east-1": DEFAULT_DNS_SUFFIX,
+        "eu-central-1": DEFAULT_DNS_SUFFIX,
+        "cn-north-1": CHINA_DNS_SUFFIX,
+        "cn-northwest-1": CHINA_DNS_SUFFIX,
+        "us-gov-west-1": DEFAULT_DNS_SUFFIX,
+        "us-isob-east-1": GOV_DNS_SUFFIX,
+        **common_regions_with_dns_suffix,
+    },
+    "emr": {
+        "fips-ca-central-1": DEFAULT_DNS_SUFFIX,
+        **common_regions_with_dns_suffix,
+    }
+}
+
+
+def register_override_ssl_common_name(cli):
+    cli.register_last(
+        'before-building-argument-table-parser',
+        update_endpoint_url
+    )
+
+def update_endpoint_url(session, parsed_globals, **kwargs):
+    service = parsed_globals.command
+    service_regions = sslCommonNameServiceRegions.get(service)
+    # only change url if user has not overrided already themselves
+    if (
+        service_regions is not None
+        and parsed_globals.endpoint_url is None
+    ):
+        region = parsed_globals.region
+        # only resolve region if user has not provided it in the command
+        if region is None:
+            region = (
+                session.get_config_variable('region')
+                or os.environ.get('AWS_DEFAULT_REGION')
+            )
+        dnsSuffix = service_regions.get(region)
+        if dnsSuffix is not None:
+            service_data = session.get_service_data(service)
+            endpointPrefix = service_data['metadata']['endpointPrefix']
+            parsed_globals.endpoint_url = f"https://{endpointPrefix}.{region}.{dnsSuffix}"
```

Comment:
Is this right? Aren't almost all of the entries supposed to be region.endpointPrefix.dnsSuffix?

I think the idea here was to hardcode most of this so we're not relying on the session anymore. If we went with this approach we could just pull the endpoint resolver off the session and all this code would go away.

Reply:
no that's what `sslCommonName` is doing for the most part currently. I was under the impression we wanted to update all services/regions using `{region}.{endpointPrefix/service}.{dnsSuffix}` to `{endpointPrefix/service}.{region}.{dnsSuffix}`. 

And sure I can remove the call to grab the service data and hard code it in the config mentioned above.

Reply:
I think the point of this PR was to preserve all the sslCommonName entries for the CLI. That allows us to remove the code path in Botocore.